flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/AutoClosableProcess.java

```diff
@@ -167,6 +186,18 @@ private static void processStream(final InputStream stream, final Consumer<Strin
 		).start();
 	}
 
+	private static void processOutputStream(final OutputStream stream, final List<String> inputs) {
+		new Thread(() -> {
+			try (PrintStream printStream = new PrintStream(stream, true, StandardCharsets.UTF_8.name())) {
```

Comment:
Actually, try with resource will close this output put stream, and in this way, will close process.
I think you should create a better method name for this, and add comments to explain.